proxy/src/source.rs

```diff
@@ -1,7 +1,31 @@
 use futures::future::Future;
 use std::collections::HashMap;
+use std::convert::{TryInto};
 
-use radicle_registry_client::{CryptoPair as _, ProjectDomain, ProjectName, H256};
+use radicle_registry_client::{
+    CryptoPair as _, ProjectDomain, ProjectName, H256, Error as RegistryError
+};
+
+type BadInput = String;
```

Comment:
I find it confusing having two `BadInput`s, this one and the variant in `SourceError`